airflow/contrib/hooks/databricks_hook.py

```diff
@@ -0,0 +1,190 @@
+        :param retry_limit: The number of times to retry the connection in case of
+            service outages.
+        :type retry_limit: int
+        """
+        self.databricks_conn_id = databricks_conn_id
+        self.databricks_conn = self.get_connection(databricks_conn_id)
+        self.timeout_seconds = timeout_seconds
+        self.retry_limit = retry_limit
+
+    def _parse_host(self, host):
+        """
+        The purpose of this function is to be robust to improper connections
+        settings provided by users, specifically in the host field.
+
+        For example -- when users supply ``https://XX.cloud.databricks.com`` as the
+        host, we must strip out the protocol to get the host.
+            _parse_host(https://XX.cloud.databricks.com) -> XX.cloud.databricks.com.
+
+        In the case where users supply the correct ``XX.cloud.databricks.com`` as the
+        host, this function is a no-op.
+            _parse_host(XX.cloud.databricks.com) -> XX.cloud.databricks.com.
+        """
+        urlparse_host = urlparse.urlparse(host).hostname
+        if urlparse_host:
+            # In this case, host = https://XX.cloud.databricks.com
+            return urlparse_host
+        else:
+            # In this case, host = XX.cloud.databricks.com
+            return host
+
+    def _do_api_call(self, endpoint_info, api_parameters):
+        """
+        Utility function to perform an API call with retries
+        :param endpoint_info: Tuple of method and endpoint
+        :type endpoint_info: (string, string)
+        :param api_parameters: Parameters for this API call.
+        :type api_parameters: dict
+        :return: If the api call returns a OK status code,
+            this function returns the response in JSON. Otherwise,
+            we throw an AirflowException.
+        :rtype: dict
+        """
+        method, endpoint = endpoint_info
+        url = 'https://{host}/{endpoint}'.format(
+            host=self._parse_host(self.databricks_conn.host),
+            endpoint=endpoint)
+        auth = (self.databricks_conn.login, self.databricks_conn.password)
+        if method == 'GET':
+            request_func = requests.get
+        elif method == 'POST':
+            request_func = requests.post
+        else:
+            raise AirflowException('Unexpected HTTP Method: ' + method)
+
+        for attempt_num in range(1, self.retry_limit+1):
+            try:
+                response = request_func(
+                    url,
+                    json=api_parameters,
+                    auth=auth,
+                    headers=USER_AGENT_HEADER,
+                    timeout=self.timeout_seconds,
+                    verify=False)
+                if response.ok:
+                    return response.json()
+                else:
+                    # In this case, the user probably made a mistake.
+                    # Don't retry.
+                    raise AirflowException(response.content)
```

Comment:
We should log that together with the content. Is it not guaranteed that the content contains status code.